### PR TITLE
Move to gcc 7.1

### PR DESCRIPTION
**Pull request by mythi**
Many fixes (including meta-ros GCC7.1 readiness) have been merged
so move back to follow OE-core default GCCVERSION.

### CODE REVIEW

File: meta-refkit-computervision/recipes-computervision/librealsense/files/0001-Add-missing-includes-for-functional.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
+From 7b1581644c5fbdec3a22cd512557b391dc74ebf0 Mon Sep 17 00:00:00 2001
+From: Till Hofmann <thofmann@fedoraproject.org>
+Date: Wed, 22 Mar 2017 14:23:24 +0100
+Subject: [PATCH] Add missing includes for <functional>
+
+Newer GCC versions (>=7) fail with an error if the header is not
+included an std::function is used.
+
+Upstream-status: Backport
```
Comment (pohly) [inline meta-refkit-computervision/recipes-computervision/librealsense/files/0001-Add-missing-includes-for-functional.patch]: Mind the spelling (Upstream-Status) and that "Backport" usually has more information about what was backported. For example:

    libxml2-fix_NULL_pointer_derefs.patch:Upstream-Status: Backport [https://git.gnome.org/browse/libxml2/commit/?id=e905f08123e4a6e7731549e6f09dadff4cab65bd]

File: meta-refkit-gateway/recipes-core/iotivity/files/0001-iotivity-adapt-to-GCC7.patch [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The needed fixes are explained in GCC7 porting guide. This
+was sufficient to fix the failures:
+
+"Previously components such as std::bind and std::function
+were implicitly defined after including unrelated headers such
+as <memory>, <future>, <mutex>, and <regex>. Correct code
+should #include <functional> to define them."
+
+Upstream-status: Pending
```
Comment (pohly) [inline meta-refkit-gateway/recipes-core/iotivity/files/0001-iotivity-adapt-to-GCC7.patch]: Spelling again.

Comment (pohly) [inline meta-refkit-gateway/recipes-core/iotivity/files/0001-iotivity-adapt-to-GCC7.patch]: Can we avoid the "Pending" status and submit this right away, before merging?

File: meta-refkit-core/conf/distro/include/refkit-config.inc [hunk 3]
```diff
@@ -47,6 +47,9 @@ REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-firewall"
 # Reconfigure several upstream recipes for the computer vision profile.
 REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-computervision"
 
+# Reconfigure/patch upstream recipes for the gateway vision profile.
+REFKIT_DEFAULT_DISTRO_FEATURES += "refkit-gateway"
```
Comment (pohly) [inline meta-refkit-core/conf/distro/include/refkit-config.inc]: This might be better to have in a separate commit. It's not mentioned in the commit message either.

**Comment by mythi** [comment]
@pohly I've addressed the review feedback. The iotivity fix was a backport after all (OTOH, it was not found in iotivity's github mirror which I looked first. after cloning the gerrit tree I noticed it).
